hphp/runtime/ext/pdo/ext_pdo.cpp

```diff
@@ -2037,7 +2038,31 @@ static bool do_fetch(sp_PDOStatement stmt,
 
     case PDO_FETCH_CLASS:
       if ((flags & PDO_FETCH_SERIALIZE) == 0 || idx) {
-        ret.toObject()->o_set(name, val);
+
+        if (ret.getObjectData()->instanceof(s_PDOEntity)) {
+          if (!stmt->support(PDOStatement::MethodGetColumnMeta)) {
+            pdo_raise_impl_error(stmt->dbh, stmt, "IM001",
+                                 "driver doesn't support meta data");
+            return false;
+          }
+
+          setPDOErrorNone(stmt->error_code);
+          Array meta;
+
+          if (!stmt->getColumnMeta(idx, meta)) {
+            PDO_HANDLE_STMT_ERR(stmt);
+            return false;
+          }
+            
+          Array params = Array::Create();
+          params.append(name);
+          params.append(val);
+          params.append(meta);
+          ret.asCObjRef()->o_invoke("setfield", params);
```

Comment:
>TODO Task #2584896: o_invoke and o_invoke_few_args are deprecated. 
>These APIs don't properly take class context into account when looking up the method ...

Where can i see task #2584896 to understand problem. `o_invoke` looks like me too easy and best helper. I can't imagine a structure to call invokeFunc method because it needs too verbose informations. Do `Func` and `Class` already belong to `ObjectData`? Why i need to determine references their seperate ways? Late static calls?

So I don't have an idea about `VarEnv* varEnv = nullptr, StringData* invName = nullptr, InvokeFlags flags = InvokeNormal`

Or can you help me for this? I allowed maintainers to edit